packages/react-select/src/Select.tsx

```diff
@@ -653,7 +669,11 @@ export default class Select<
       const focusedValue = clearFocusValueOnUpdate
         ? getNextFocusedValue(state, selectValue)
         : null;
-      const focusedOption = getNextFocusedOption(state, focusableOptions);
+      const focusedOption = getNextFocusedOption(
+        state,
+        focusableOptions,
+        prevProps
```

Comment:
Ensuring we always get a value, even when the prop doesn't exist

```suggestion
        !!prevProps?.alwaysFocusFirstMenuOption
```